x-pack/plugins/security_solution/public/network/containers/network_http/index.tsx

```diff
@@ -35,126 +33,168 @@ export interface NetworkHttpArgs {
   ip?: string;
   inspect: inputsModel.InspectQuery;
   isInspected: boolean;
-  loading: boolean;
   loadPage: (newActivePage: number) => void;
   networkHttp: NetworkHttpEdges[];
   pageInfo: PageInfoPaginated;
   refetch: inputsModel.Refetch;
   totalCount: number;
 }
 
-export interface OwnProps extends QueryTemplatePaginatedProps {
-  children: (args: NetworkHttpArgs) => React.ReactNode;
+interface UseNetworkHttp {
+  id?: string;
   ip?: string;
   type: networkModel.NetworkType;
+  filterQuery?: ESTermQuery | string;
+  endDate: string;
+  startDate: string;
+  skip: boolean;
 }
 
-export interface NetworkHttpComponentReduxProps {
-  activePage: number;
-  isInspected: boolean;
-  limit: number;
-  sort: NetworkHttpSortField;
-}
+export const useNetworkHttp = ({
+  endDate,
+  filterQuery,
+  id = ID,
+  ip,
+  skip,
+  startDate,
+  type,
+}: UseNetworkHttp): [boolean, NetworkHttpArgs] => {
+  const getHttpSelector = networkSelectors.httpSelector();
+  const { activePage, limit, sort } = useSelector(
+    (state: State) => getHttpSelector(state, type),
+    shallowEqual
+  );
+  const { data, notifications, uiSettings } = useKibana().services;
+  const refetch = useRef<inputsModel.Refetch>(noop);
+  const abortCtrl = useRef(new AbortController());
+  const defaultIndex = uiSettings.get<string[]>(DEFAULT_INDEX_KEY);
+  const [loading, setLoading] = useState(false);
+
+  const [networkHttpRequest, setHostRequest] = useState<NetworkHttpRequestOptions>({
+    defaultIndex,
+    factoryQueryType: NetworkQueries.http,
+    filterQuery: createFilter(filterQuery),
+    ip,
+    pagination: generateTablePaginationOptions(activePage, limit),
+    sort: sort as SortField,
```

Comment:
I have the same issue of SortField too, as some tables do not have any sorting,
what about we make `field` and `direction` optional in SortField?

Reply:
per our conversation, we have decided to refactor that at the end